service/namespace.go

```diff
@@ -12,37 +12,37 @@ const eventChannel string = "Event"
 const taskChannel string = "Task"
 const resultChannel string = "Result"
 
-// namespace returns the namespace of the service.
-func (service *Service) namespace() []string {
+// Namespace returns the namespace of the service.
+func (service *Service) Namespace() []string {
```

Comment:
I think namespace is mostly used for all docker related functions so this should be moved directly to the manager package instead of being exposed.

Reply:
it can be used by Kubernetes or any other implementation of Manager to identify a mesg service.

Reply:
but they will have different restrictions, here for example we are hashing in sha1 to make sure that the name will not be too long for docker, maybe kubernetes doesn't have this limitation or don't have a namespace system at all

Reply:
i mean, we'll still need to assign some unique names to services in any orchestration tool that's implemented. we can just have one common unique name creator in service pkg and use it everywhere like this.

we can also change this `Namespace()` func to commonly return a string instead and Manager implementations can use it the way how it fits for themselves. tho, i'm not pushing to have this func here. we can refactor this in future if needed or in an another PR. i'm all ok about it

Reply:
There is already the `service.Hash` that can be use to get a unique name for a service.
I agree with Anthony, we should move it in Manager.

Reply:
Let's continue the conversation here: https://github.com/mesg-foundation/core/issues/957